Tests/RoslynPluginGeneratorTests/AnalyzerPluginGeneratorTests.cs

```diff
@@ -0,0 +1,361 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="AnalyzerPluginGeneratorTests.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+using System;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarQube.Plugins.Test.Common;
+using System.IO;
+using NuGet;
+using System.Linq;
+using System.Collections.Generic;
+namespace SonarQube.Plugins.Roslyn.RoslynPluginGeneratorTests
+{
+    /// <summary>
+    /// Tests for NuGetPackageHandler.cs
+    /// </summary>
+    [TestClass]
+    public class AnalyzerPluginGeneratorTests
+    {
+        public TestContext TestContext { get; set; }
+
+        private enum License { Required, NotRequired };
+        private enum Node { A, B, C, D, E, F };
```

Comment:
More descriptive names would be better e.g. "Parent", "Child1", "Child2", "GrandChild1_1"... or "Level1", "Level2a", "Level2b".